src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java

```diff
@@ -59,10 +65,25 @@ public SpatialViewerDatasetService(
 	}
 
   public List<SpatialViewerDataset> getSpatialViewerDataset() throws Exception {
-    List <SpatialViewerDataset> datasets = new ArrayList<>();
-    datasets.addAll(externalLinkRepo.findAll());
+    List <SpatialViewerDataset> datasetsFinal = new ArrayList<>();
+    List <SpatialViewerFileDataset> datasets = new ArrayList<>();
+    Map<String, SpatialViewerFileDataset> fileMap = new HashMap<>();
+    Double maxReleaseVersion = fileRepo.max();
     datasets.addAll(fileRepo.findAll());
-    return datasets;
+    for (SpatialViewerFileDataset spatialViewerFileDataset : datasets){
+        if(spatialViewerFileDataset.getReleaseVersion() != null){
+            if (Double.compare(spatialViewerFileDataset.getReleaseVersion(), maxReleaseVersion) == 0){
+                spatialViewerFileDataset.setReleaseVersionDisplay("Recently Released");
```

Comment:
This should be extracted to a constant. Remember constants should be `private static final`, and live at the top of the file with other class variables.